src/store/todoSlice/todoSlice.test.js

```diff
@@ -0,0 +1,151 @@
+import reducer, {
+  addTodo,
+  deleteTodo,
+  editTodo,
+  toogleTodo,
+} from "./todoSlice";
+
+test("should handle a todo being added to an empty list", () => {
+  const previousState = {
+    todos: [],
+  };
+
+  expect(
+    reducer(
+      previousState,
+      addTodo({
+        id: 0,
+        title: "Run the tests",
+        completed: false,
+      })
+    )
+  ).toEqual({
+    todos: [
+      {
+        id: 0,
+        title: "Run the tests",
+        completed: false,
+      },
+    ],
+  });
+});
+
+test("should handle a todo being added to an existing list", () => {
```

Comment:
rewrite all tests to  given-when-then template https://www.agilealliance.org/glossary/gwt/#q=~(infinite~false~filters~(postType~(~'page~'post~'aa_book~'aa_event_session~'aa_experience_report~'aa_glossary~'aa_research_paper~'aa_video)~tags~(~'given*20when*20then))~searchTerm~'~sort~false~sortDirection~'asc~page~1)